c/kore/hello/src/hello.c

```diff
@@ -0,0 +1,43 @@
+#include <kore/kore.h>
+#include <kore/http.h>
+
+int		home(struct http_request *);
+int		user(struct http_request *);
+int		user_details(struct http_request *);
+
+int
+home(struct http_request *req)
+{
+	http_response(req, 200, NULL, 0);
+	return (KORE_RESULT_OK);
+}
+
+int
+user(struct http_request *req)
+{
+	http_response(req, 200, NULL, 0);
+	return (KORE_RESULT_OK);
+}
+
+int
+user_details(struct http_request *req)
+{
+	char		*id;
+	size_t		len;
+	struct kore_buf		*buf;
+	u_int8_t		*data;
+
+	http_populate_get(req);
+
+	buf = kore_buf_alloc(1024);
+
+	if (http_argument_get_string(req, "id", &id))
+		kore_buf_appendf(buf, id);
+
+	data = kore_buf_release(buf, &len);
+
+	http_response(req, 200, data, len);
```

Comment:
Again, shouldn't there be a way to pass stack data in to this or moving the stack data instead of passing a slow HEAP allocated pointer?  I havn't looked in depth in to kore but I wouldn't expect any half decent C/C++ web framework to use heap allocations like this...

Reply:
OK. They are copied from the examples of Kore. I am not familiar with these yet.
For example:
* https://github.com/jorisvink/kore/blob/3.1.0-release/examples/generic/src/example.c
* https://github.com/jorisvink/kore/blob/3.1.0-release/examples/parameters/src/parameters.c
* ...

Reply:
The API docs state that you can allocate the buffer on the stack, I didn't see relevant code for it, but it does support it.